### ISSUE TITLE
NameError: name '_VOoM2657' is not defined

### ISSUE THREAD

**Issue by Sangdol** [vim-voom/VOoM#11]
Hello,

First of all, thanks for the awesome plugin.

After I upgraded my vim to 8.1.150,  I get this error when I try to run VOoM. 

> Error detected while processing .../VOoM/autoload/voom.vim:
> line   92:
> /must>not&exist/foo:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
> Error detected while processing function voom#Help[48]..voom#Init:
> line   46:
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
> NameError: name '_VOoM2657' is not defined

I've tried reinstalling but it didn't help. 

Can someone help me?

Thanks.

**Comment by vim-voom** [comment]
Please check if Python scripting works properly in your Vim.
If the following commands don't work then the problem is with your Vim
version and/or Python, not VOoM plugin.

Check if Vim has Python 2 support:
    :py print 2**0.5
    :py import sys; print sys.version

Check if Vim has Python 3 support:
    :py3 print(2**0.5)
    :py3 import sys; print(sys.version)

**Comment by vim-voom** [comment]
There is such issue reported for macOS:
https://github.com/vim/vim/issues/3117

A silly workaround suggested there: insert on top of .vimrc

if has('python3')
  silent! python3 1
endif

I cannot test this because I don't have macOS. If it does help, you
should also run VOoM test suite from my VOoM_extras repo to make sure
there are no other issues.

**Comment by Sangdol** [comment]
Thanks a lot for the answer. I just found out that Python 2 is not supported in the new version for some reason. Does vim have to support both Python2 and Python3 to run VOoM?

**Comment by vim-voom** [comment]
On 7/17/18, Sanghyun Lee <notifications@github.com> wrote:
> Thanks a lot for the answer. I just found out that Python 2 is not supported
> in the new version for some reason. Does vim have to support both Python2
> and Python3 to run VOoM?

VOoM will work with either Python2 or Python3. This is since version
v5.2. You can read about it at
http://www.vim.org/scripts/script.php?script_id=2657
(install details) on in help file.

**Comment by Sangdol** [comment]
I've tested a bit more.

The issue you shared was just about this message. 

    /must>not&exist/foo:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses

When I tried this 

```
if has('python3')
  silent! python3 1
endif
```

the message disappeared but VOoM still didn't work.

So the difference between my previous version of vim and the new version was the version of Python. In my previous version, vim was using Python2 and in the new version it was using Python3. 

I've tried re-installing vim with Python2 and it worked. And the version of VOoM is 5.3.
![image](https://user-images.githubusercontent.com/803223/42902291-6f2a0640-8ace-11e8-8187-333cb1e675a4.png)

Is there any posibility that VOoM 5.3 is not compatible with Python3?

**Comment by vim-voom** [comment]
> Is there any posibility that VOoM 5.3 is not compatible with Python3?


The issue is with Vim and new Python 3.7. No problems with Pythons 3.4,
3.5, 3.6. I've reproduced this issue on Windows using gvim 8.1.0191 from
https://tuxproject.de/projects/vim/ and Python 3.7.0. Python warning
DeprecationWarning emitted by Vim internals does not break VOoM plugin,
but it breaks another plugin that also uses Python. The
    silent! python3 1
trick  works and there are no other issues. I don't know why it does not
work for you. Try putting it in autoload/voom.vim in addition to .vimrc.

If you use some fancy plugin manager, try removing and installing VOoM
manually just in case. Delete all voom* stuff in ~/.vim first.

**Comment by grem11n** [comment]
Workaround with

```
if has('python3')
  silent! python3 1
endif
```

worked for me with macOS 10.13.6 (Darwin Kernel Version 17.7.0) and Python 3.7.0

**Comment by vim-voom** [comment]
This issue was fixed by Vim patch 8.1.0201.

**Comment by bfedie518** [comment]
This problem has popped back up when using Python 3.8.3 and Vim 8.2 on Windows 10, but the workaround works.
